### PR TITLE
(#16490)[package] nas: Fix recipe on ARM platforms

### PR DESCRIPTION
**Pull request by dariusarnold**
The repository contained an old version of the config.guess and config.sub files, which are not compatible with building on ARM. During the configure phase it is unable to guess the system type. Overwriting both files with a new version fixes that and makes it recognize the system correctly.

Specify library name and version:  **nas/1.9.5**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/nas/all/conanfile.py [hunk 1]
```diff
@@ -85,6 +87,10 @@ def _imake_make_args(self):
     def build(self):
         apply_conandata_patches(self)
 
+        config_folder = os.path.join(self.source_folder, "config")
+        shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB, os.path.join(config_folder, "config.sub"))
+        shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS, os.path.join(config_folder, "config.guess"))
```
Comment (SpaceIm) [inline recipes/nas/all/conanfile.py:92]:
```suggestion

            for gnu_config in [
                self.conf.get("user.gnu-config:config_guess", check_type=str),
                self.conf.get("user.gnu-config:config_sub", check_type=str),
            ]:
                if gnu_config:
                    config_folder = os.path.join(self.source_folder, "config")
                    copy(self, os.path.basename(gnu_config), src=os.path.dirname(gnu_config), dst=config_folder)
```

File: recipes/nas/all/conanfile.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 from conan.tools.files import chdir, get, download, export_conandata_patches, apply_conandata_patches, rm, copy
 from conan.tools.gnu import AutotoolsToolchain, Autotools, AutotoolsDeps
 import os
+import shutil
```
Comment (SpaceIm) [inline recipes/nas/all/conanfile.py:8]:
```suggestion
```

**Comment by dariusarnold** [comment]
Should the changes be squashed?

**Comment by dariusarnold** [comment]
What should I do about the two failed CI jobs? I can't get any details on them since I am not allowed to view the builds (error 403).

**Comment by StellaSmith** [comment]
> Should the changes be squashed?

The bot will squash all commits when merging.

> What should I do about the two failed CI jobs? I can't get any details on them since I am not allowed to view the builds (error 403).

You can close and reopen the pull request to trigger C3I  again.

**Comment by prince-chrismc** [comment]
You need to comment in #4 for it to build and get accepted

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by samuel-emrys — APPROVED** [review]

**Pull request merged**